grumphp.yml

```diff
@@ -3,40 +3,83 @@ grumphp:
     EXEC_GRUMPHP_COMMAND: docker-compose run -e'PHP_ERROR_REPORTING=E_ALL & ~E_DEPRECATED' --rm -T php
   extensions:
     - GrumphpDrupalCheck\ExtensionLoader
+    - Metadrop\PhpCompatibilityTask\ExtensionLoader
   hooks_dir: ~
+  hooks_preset: local
   tasks:
     git_commit_message:
       enforce_no_subject_trailing_period: true
       max_subject_width: 130
       case_insensitive: false
+      matchers:
+        Must contain gitlab reference (if no reference use 0): '/(^Issue #[0-9]+:|^Merge branch.)/'
+        Space and uppercase after reference: '/(^Issue #[0-9]+: [A-Z]|^Merge branch.)/'
+    git_blacklist:
+      keywords:
+        - "var_dump"
+        - "print_r"
+        - "console.log"
+      whitelist_patterns: [ ]
     # Adjust here the format that must have
     # the git branch names of your project.
     git_branch_name:
       whitelist:
-        - "feature/[0-9A-Za-z\\s_-]*"
-        - "hotfix/[0-9A-Za-z\\s_-]*"
-        - "bugfix/[0-9A-Za-z\\s_-]*"
-        - "develop"
+        - "feature\/[0-9A-Za-z\\s_-]*"
+        - "bugfix\/[0-9A-Za-z\\s_-]*"
+        - "hotfix\/[0-9A-Za-z\\s_-]*"
       blacklist:
-        - "release"
-        - "main"
+        - develop
```

Comment:
I would allow committing to the _develop_ branch directly, there are some special cases when it is strongly needed: small fixes, project first steps.

Reply:
I am worried that:
-  if developers are forced to use -n to commit into develop we may accidentally introduce codings standard errors (for example).
- There are cases where a huge amount of commits are needed to add into dev.  It may be uncomfortable to force ignoring every commit.

Reply:
Before you use -n you should try to commit without -n to check for codings standard. I think committing with -n directly to any branch is a bad practice that nobody should do. 95% of the development time, develop branch should be blocked so I think that this should be the standard behaviour. In the extreme case where you have to force commit a lot to develop, you can alter locally (and temporary) the grumphp file.

Reply:
- Altering a configuration file for a specific moment is a bad practice because devs can accidentally commit that file. Files tracked by git should only be modified if those changes are needed to be committed.
- I didn't say devs will commit without adding -n without committing correctly first. When I say 'forced' I mean that developers are required to do that to commit into develop. Otherwise, you can't commit into develop. So devs are forced. 
- As devs are forced, a dev can accidentally use -n for every commit into develop. We may have errors as we are humans. As they accidentally commit to develop, they may introduce errors.

Sorry but I think that :
- If there are use cases where it is needed to commit straight into a branch, that branch must not be blacklisted. Only branches where there are no need to commit straight into must be blacklisted.
- Commits should be always done without -n, and the use cases when it is needed to use -n  must be very limited. This change is enlarging the use cases.

Reply:
I think it's safe to say that we agree to disagree :)  I like to be strict with validations because as you say we are humans, so I'm still thinking that develop must be blacklisted. But, in order to not block this pull request, I will whitelist develop and if in the future we see that it could be a good option to move it to the blacklist, we will do so. I don't think it's a blocking issue, even if it makes me uneasy.